### PR TITLE
New contribution docs

### PR DESCRIPTION
**Pull request by mhils**
- [x] Remove old cruft from the docs
- [x] Add new contributing section to the readme

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -95,26 +85,23 @@ Testing
...
-The following tox environments are relevant for local testing:
+For speedier testing, we recommend you run `py.test`_ directly on individual test files or folders:
 
 .. code-block:: text
 
-    tox -e py35  # runs all tests with Python 3.5
-    tox -e docs  # runs a does-it-compile check on the documentation
-    tox -e lint  # runs the linter for coding style checks
+    cd test/mitmproxy/addons
+    py.test --cov mitmproxy.addons.anticache --looponfail test_anticache.py
```
Comment (Kriechi) [inline README.rst]: The "official" usage is actually spelled `pytest` when invoked via CLI.
At least thats what the py.test documentation is using.

Comment (mhils) [inline README.rst]: Oh, looks like they actually changed it. Good catch, thanks. 😃

File: docs/dev/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+Contributing
+============
+
+As an open source project, **mitmproxy** welcomes contributions of all forms. 
+
+Please head over to the README_ to get started! 😃
+
+
+.. _README: https://github.com/mitmproxy/mitmproxy/blob/master/README.rst
```
Comment (Kriechi) [inline docs/dev/contributing.rst]: Should we mention & link the GitHub issue label for "good-first-contribution"?

File: README.rst [hunk 3]
```diff
@@ -49,42 +49,32 @@ The installation instructions are `here <http://docs.mitmproxy.org/en/stable/ins
 If you want to contribute changes, keep on reading.
 
 
-Hacking
--------
+Setting Up a Development Environment
```
Comment (Kriechi) [inline README.rst]: Ho about: `s/Setting Up a //`?

File: README.rst [hunk 4]
```diff
@@ -95,26 +85,23 @@ Testing
...
-
-Please ensure that all patches are accompanied by matching changes in the test
-suite. The project tries to maintain 100% test coverage.
-
-You can also use `tox` to run the full suite of tests, including a quick test
-to check documentation and code linting.
+    tox
 
-The following tox environments are relevant for local testing:
+For speedier testing, we recommend you run `py.test`_ directly on individual test files or folders:
```
Comment (Kriechi) [inline README.rst]: Not sure how other people are using it, but I almost always invoke pytest via tox - just to have a clean env for testing: e.g.`tox -e py35 -- test/mitmproxy/test_foo.py -s -x -v`

File: docs/dev/contributing.rst (hunk 2 above)
Comment (mhils) [inline docs/dev/contributing.rst]: Going to add a nontechnical section to the README next, which will include this 👍

File: README.rst (hunk 4 above)
Comment (mhils) [inline README.rst]: Should not matter too much as this is happening within the venv anyway. Let's keep the simplified version for now.

**Comment by mhils** [comment]
Done. Do we want to add anything else to the "Contributing" section?

File: README.rst [hunk 5]
```diff
@@ -48,73 +48,64 @@ Installation
...
-For convenience, the project includes an autoenv_ file (`.env`_) that
-auto-activates the virtualenv when you cd into the mitmproxy directory.
-
-
 Testing
 -------
 
 If you've followed the procedure above, you already have all the development
-requirements installed, and you can simply run the test suite:
+requirements installed, and you can run the full test suite (including tests for code style and documentation) with tox_:
```
Comment (Kriechi) [inline README.rst]: ... Make sure to have Python executables for all supported Python releases available so tox can find and use them.

Comment (mhils) [inline README.rst]: I think it's fine to just test on one Python version for regular development, CI will catch all the obscure issues. Tox does a very good job at communicating its results:
```
ERROR:   py35: InterpreterNotFound: python3.5
  py36: commands succeeded
  docs: commands succeeded
  lint: commands succeeded
```

**Pull request merged**